src/btree/bt_cursor.c

```diff
@@ -514,6 +514,16 @@ __wt_btcur_reset(WT_CURSOR_BTREE *cbt)
 
     WT_STAT_CONN_DATA_INCR(session, cursor_reset);
 
+    /* Clear bounds if they are set. */
```

Comment:
This seems like the right location to me but I've created a ticket to follow up later in the project, there are a few other potential functions for it so I want to make sure its in the right place. Theres no additional work right now here.